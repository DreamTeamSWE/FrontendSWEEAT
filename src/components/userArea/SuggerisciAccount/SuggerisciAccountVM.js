import { useState } from "react";
import { suggest } from "../../../services/services";

export default function AreaPersonaleVM() {
  const [stato, setStato] = useState("");
  const [accountId, setAccountId] = useState("");

  const suggerisciClick = async () => {
    const message = await suggest(accountId);
    setStato(message);
  };

  const inputHandler = (event) => {
    setAccountId(event.target.value);
  };

  return {
    stato,
    accountId,
    suggerisciClick,
    inputHandler,
  };
}
