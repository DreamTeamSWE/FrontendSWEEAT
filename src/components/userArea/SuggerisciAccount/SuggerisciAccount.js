import React, { useState } from "react";

import SuggerisciAccountVM from "./SuggerisciAccountVM";

function Suggerisci() {
  const { stato, accountId, suggerisciClick, inputHandler } =
    SuggerisciAccountVM();

  return (
    <div className="suggerisci">
      <h1>Suggerisci account</h1>
      <input type="text" value={accountId} onChange={inputHandler} />
      <button onClick={suggerisciClick}>Suggerisci account</button>
      {stato && <h4>{stato}</h4>}
    </div>
  );
}

export default Suggerisci;
