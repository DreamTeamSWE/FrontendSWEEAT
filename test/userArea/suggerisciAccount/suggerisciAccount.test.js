import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import SuggerisciAccount from '../../../src/components/userArea/SuggerisciAccount/SuggerisciAccount';
import {render, screen} from '@testing-library/react';
import {BrowserRouter as Router} from 'react-router-dom';
import '@testing-library/jest-dom';

configure({adapter: new Adapter()});

describe('Check render controllo link mobile', () => {
  beforeEach(() =>{
  render(
    <Router>
      <SuggerisciAccount />
    </Router>
  )});

  test('Check render results', () =>{
    const sa = shallow(<SuggerisciAccount/>);
    expect(sa).toMatchSnapshot();
  })

});
