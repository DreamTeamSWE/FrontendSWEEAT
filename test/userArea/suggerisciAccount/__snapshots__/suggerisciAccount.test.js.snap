// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check render controllo link mobile Check render results 1`] = `ShallowWrapper {}`;
